src/modules/menu/components/common/SearchableDropdown.vue

```diff
@@ -0,0 +1,128 @@
+<script setup>
+import { computed, ref, useSlots } from 'vue'
+const slots = useSlots()
+import SearchList from '@/modules/menu/components/common/SearchList.vue'
+
+const emit = defineEmits(['update:modelValue'])
+
+const props = defineProps({
+    modelValue: {
+        type: [Number, String],
+        required: true,
+    },
+    entries: {
+        type: Array,
+        default: () => [],
+    },
+})
+
+const isDropdownOpen = ref(false)
+const inputValue = ref('')
+// reference to the component
+const searchList = ref(null)
+
+const displayEntry = computed({
+    get() {
+        if (isDropdownOpen.value) {
+            return inputValue.value
+        } else {
+            return props.modelValue
+        }
+    },
+    set(value) {
+        inputValue.value = value
+    },
+})
+
+const dropdownEntries = computed(() => {
+    return props.entries
+        .filter((entry) => {
+            if (inputValue.value) {
+                return entry.toString().includes(inputValue.value)
+            } else {
+                // nothing searching, return all
+                return true
+            }
+        })
+        .map((entry) => ({
+            htmlDisplay: entry.toString(),
+            emphasize: inputValue.value?.toString() || '',
+            url: entry,
+        }))
+})
+
+const isSlotUsed = computed(() => {
+    return !!slots.default
+})
+
+function toggleList() {
+    if (isDropdownOpen.value) {
+        closeList()
+    } else {
+        openList()
+    }
+}
+
+function openList() {
+    isDropdownOpen.value = true
+}
+
+function closeList() {
+    isDropdownOpen.value = false
+    inputValue.value = null
+}
+
+function chooseEntry(value) {
+    emit('update:modelValue', value)
+    closeList()
+}
+
+function focusSearchlist() {
+    if (inputValue.value) {
+        searchList.value.goToFirst()
+    } else {
+        searchList.value.goToSpecific(props.modelValue)
+    }
+}
+</script>
+
+<template>
+    <div v-click-outside="closeList">
+        <div class="d-flex">
+            <form action="" class="input-group input-group-append" data-cy="searchable-dropdown">
+                <input
+                    :value="displayEntry"
+                    class="form-control rounded-end-0"
+                    @input="inputValue = $event.target.value"
+                    @focusin="openList"
+                    @keydown.esc.prevent="closeList"
+                    @keydown.down.prevent="focusSearchlist"
+                />
+
+                <div class="input-group-append btn-group btn-group-sm">
```

Comment:
Might be a better idea to use bigger button for touch device, so either the btn-group or btn-group-lg. Now especially the dropdown button is a bit small and difficult to touch on mobile.